var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -115,6 +116,7 @@ class Libfabric(AutotoolsPackage):
     depends_on("opa-psm2", when="fabrics=psm2")
     depends_on("psm", when="fabrics=psm")
     depends_on("ucx", when="fabrics=mlx")
+    depends_on("ucx", when="fabrics=ucx")
```

Comment:
Can you add a `when=@1.18.0:` clause to make sure only versions newer than 1.18 can be built with this provider?